wolfcrypt/src/aes.c

```diff
@@ -2286,13 +2351,49 @@ int wc_AesSetIV(Aes* aes, const byte* iv)
                                                       PIC32_CRYPTOALGO_RCBC );
         return 0 ;
     }
-
+    #ifdef HAVE_AES_DECRYPT
     int wc_AesCbcDecrypt(Aes* aes, byte* out, const byte* in, word32 sz)
     {
         wc_AesCrypt(aes, out, in, sz, PIC32_DECRYPTION, PIC32_ALGO_AES,
                                                       PIC32_CRYPTOALGO_RCBC);
         return 0 ;
     }
+    #endif /* HAVE_AES_DECRYPT */
+#elif defined(WOLFSSL_NRF51_AES)
+    int wc_AesCbcEncrypt(Aes* aes, byte* out, const byte* in, word32 sz)
+    {
+        int i, ret;
+        int offset = 0;
+        int len = sz;
+
+        byte *iv;
+        byte temp_block[AES_BLOCK_SIZE];
+
+        iv      = (byte*)aes->reg;
+
+        while (len > 0)
```

Comment:
I think using the basic software one as a template would be better here, i.e.,

word32 blocks = sz / AES_BLOCK_SIZE;

while (blocks--)

to prevent the bad cases where len isn't an AES_BLOCK_SIZE multiple or sz > 2GB.